### ISSUE TITLE
Converting msh to vtk

### ISSUE THREAD

**Issue by Tsunyshka** [nschloe/meshio#847]
Hi!
I am trying to convert .msh (gmsh ascii) to .vtk file format with the simple comand:

meshio-convert input.msh output.vtk

But i getting strange result in my output file...
What should i do to fix this?

Attach input and output files
[mesh.zip](https://github.com/nschloe/meshio/files/4798914/mesh.zip)

**Comment by Tsunyshka** [comment]
fixed it with  --output-format vtk --ascii

**Comment by nschloe** [comment]
It should work either way. Reopening.

**Comment by Tsunyshka** [comment]
Hi!
Is it possible to use meshio-convert in a .py script?

**Comment by Tsunyshka** [comment]
> Hi!
> Is it possible to use meshio-convert in a .py script?

I’m trying to use smth like this, but it actually doesn’t work...
` import meshio 

in_file = "/path/to/test_mesh.msh"
out_file = "/path/to/test_mesh.ply"
mesh = meshio.read(in_file, file_format="msh")  # works
mesh.write(out_file, file_format="ply")  # error
 `

**Comment by gdmcbain** [comment]
Try
```python
meshio.write(out_file, mesh)
```

**Comment by nschloe** [comment]
@keileg Do you have an idea how to fix this? I'm getting
```
AssertionError: Incompatible cell data. 157 cell blocks, but 'gmsh:physical' has 16 blocks.
```
trying to read OP's input msh.

**Comment by keileg** [comment]
I think you need to store data on which CellBlock `'gmsh:physical'` belongs to; right now I think there is a (tacit) assumption that length of `'gmsh:physical'` equals the number of CellBlocks. I actually thought that gmsh only generates cells for objects decleared physical, but that may be a wrong or outdated assumption on my part.

**Comment by keileg** [comment]
I don't know if the best is to move 'gmsh:physical' to `Mesh.cell_sets`?

**Comment by nschloe** [comment]
>  I actually thought that gmsh only generates cells for objects decleared physical, but that may be a wrong or outdated assumption on my part.

@keileg I thought so too, but working with the Gmsh API it seems that this isn't the case. The data isn't really set data either. (This would be a list of indices into the points or cells array.) I would expect `cell_data["gmsh:physical"]` to list `None` for whenever a cell block as no associated data, such that `len(cells) == len(datum)` for all `cell_data`.

**Comment by keileg** [comment]
I agree, but would it not break the idea that cell_data are numpy arrays with one item per cell?

**Comment by nschloe** [comment]
Well, if there is no data for a particular cell type, we have to do _something_. Setting it to `None` makes pretty clear what's going on I think.

**Comment by keileg** [comment]
As long as it does not violate the assumptions in read/write functions, I think `None` is a good option.

**Comment by ashfaque-tk** [comment]
If .msh has two different regions, whether converting to the vtu format keeps or does not keep the tag name of the regions?
